handler/src/test/java/io/netty/handler/ssl/JdkSslEngineTest.java

```diff
@@ -279,6 +282,31 @@ public void testAlpnNoCompatibleProtocolsServerHandshakeFailure() throws Excepti
     }
 
     @Test
+    public void testAlpnCompatibleProtocolsDifferentClientOrder() throws Exception {
+        try {
+            // Typical code will not have to check this, but will get a initialization error on class load.
+            // Check in this test just in case we have multiple tests that just the class and we already ignored the
+            // initialization error.
+            if (!JdkAlpnSslEngine.isAvailable()) {
+                throw new RuntimeException("ALPN not on classpath");
+            }
+            // Even the preferred application protocol appears second in the client's list, it will be picked
+            // because it's the first one on server's list.
+            JdkApplicationProtocolNegotiator clientApn = new JdkAlpnApplicationProtocolNegotiator(false, false,
+                FALLBACK_APPLICATION_LEVEL_PROTOCOL, PREFERRED_APPLICATION_LEVEL_PROTOCOL);
+            JdkApplicationProtocolNegotiator serverApn = new JdkAlpnApplicationProtocolNegotiator(true, true,
+                PREFERRED_APPLICATION_LEVEL_PROTOCOL, FALLBACK_APPLICATION_LEVEL_PROTOCOL);
+            mySetup(serverApn, clientApn);
+           assertNull(serverException);
+           runTest(PREFERRED_APPLICATION_LEVEL_PROTOCOL);
```

Comment:
I have 2 spaces configured in Eclipse and probably added that by hand and missed 1 :) 
Funny thing is that the checkstyle plugin didn't complain at build-time. Anyway, I will start using IntelliJ for Netty development and grab the policy from your site ;-) I will fix that.